### PR TITLE
Email verification

### PR DESCRIPTION
**Pull request by remioluwatomi**
​
## Description

This PR ensures that users must have a verified email before performing certain actions by enforcing email verification checks in middleware. It also updates test cases to properly handle email verification when creating users.  

​
## Related Issue (Link to Github issue)

Feat #620 
​
## Motivation and Context

Additional Security and Authentication Layer -> This PR addresses that by ensuring users have the `email_verified_at` field set. Additionally, it reinforces middleware checks for verified users.  

​
## How Has This Been Tested?

- Created test users with `email_verified_at` set.  
- Ran the full test suite to confirm that previously failing tests now pass.  
- Ensured that authentication and authorization behaviors remain unchanged apart from the intended verification requirement.
-Modified previous test to factor in the new Email Verification system. This test include:
    - InvitationTest.php
    - UserDeletionTest.php
​

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/Http/Middleware/EnsureEmailIsVerified.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        // Ensure user is authenticated
+        dump($request->user);
```
Comment (bhimbho) [inline app/Http/Middleware/EnsureEmailIsVerified.php]: not great

File: app/Http/Controllers/Api/V1/Auth/VerificationController.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use App\Http\Controllers\Controller;
+use Illuminate\Foundation\Auth\EmailVerificationRequest;
+use Illuminate\Http\Request;
+
+class VerificationController extends Controller
+{
+    //
+
+
+    public function emailNotice($request)
```
Comment (bhimbho) [inline app/Http/Controllers/Api/V1/Auth/VerificationController.php:14]: incomplete logic

**Review by bhimbho — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Api/V1/Auth/VerificationController.php (hunk 2 above)
Comment (samuelogboye) [inline app/Http/Controllers/Api/V1/Auth/VerificationController.php:14]: in addition, what is emailNotice, notice for what?

File: tests/Feature/InvitationTest.php [hunk 3]
```diff
@@ -29,6 +29,11 @@ public function setUp(): void
             'last_name' => 'User',
         ]);
 
+        //pseudo email verification
+        $user = User::where('email', 'test@example.com')->first();
+        $user->email_verified_at = now();
+        $user->save();
+
```
Comment (samuelogboye) [inline tests/Feature/InvitationTest.php:36]: Include more test edge cases, such as expired verification links or invalid tokens.

File: app/Http/Controllers/Api/V1/Auth/VerificationController.php (hunk 2 above)
Comment (bhimbho) [reply app/Http/Controllers/Api/V1/Auth/VerificationController.php:14]: its a thing with laravel mustVerifyEmail trait. its a fallback

File: tests/Feature/InvitationTest.php (hunk 3 above)
Comment (remioluwatomi) [reply tests/Feature/InvitationTest.php:36]: Laravel requires three routes for email verification - according to the docs

- a notice route - This route is used to notify user that they have an unverified email. Also the route that laravels calls when they try to access a protected endpoint with an unverified email.

- verify email route -  This is the route that implements the actual verification of the email

- resend email route - this is needed incase they don't receive the original email or they misplace it

the notice route is required by laravel - i named the handler emailNotice.

you can check the official laravel docs: https://laravel.com/docs/11.x/verification

**Review by bhimbho — APPROVED** [review]
